doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -0,0 +1,345 @@
+        Optional<Integer> earliestYear = Optional.ofNullable(document.getPublishedPrint())
+            .flatMap(CrossrefDate::extractEarliestYear);
+
+        return earliestYear.map(this::toDate);
+    }
+
+    protected List<Contributor> toContributors(List<CrossrefAuthor> authors) {
+        List<Contributor> contributors = Collections.emptyList();
+        if (authors != null) {
+            List<Try<Contributor>> contributorMappings =
+                IntStream.range(0, authors.size())
+                    .boxed()
+                    .map(attempt(index -> toContributor(authors.get(index), index + 1)))
+                    .collect(Collectors.toList());
+
+            reportFailures(contributorMappings);
+            contributors = successfulMappings(contributorMappings);
+        }
+        return  contributors;
+    }
+
+    private List<Contributor> successfulMappings(List<Try<Contributor>> contributorMappings) {
+        return contributorMappings.stream()
+            .filter(Try::isSuccess)
+            .map(Try::get)
+            .collect(Collectors.toList());
+    }
+
+    @JacocoGenerated
+    private void reportFailures(List<Try<Contributor>> contributors) {
+        contributors.stream().filter(Try::isFailure)
+            .map(Try::getException)
+            .forEach(e -> logger.error(e.getMessage(), e));
+    }
+
+    /**
+     * Coverts an author to a Contributor (from external model to internal).
+     *
+     * @param author              the Author.
+     * @param alternativeSequence sequence in case where the Author object does not contain a valid sequence entry
+     * @return a Contributor object.
+     * @throws MalformedContributorException when the contributer cannot be built.
+     */
+    private Contributor toContributor(CrossrefAuthor author, int alternativeSequence) throws
+                                                                                      MalformedContributorException {
+        Identity identity =
+            new Identity.Builder().withName(toName(author.getFamilyName(), author.getGivenName())).build();
+        return new Contributor.Builder().withIdentity(identity)
+            .withSequence(parseSequence(author.getSequence(), alternativeSequence)).build();
+    }
+
+    /**
+     * Parses the "sequence" field of the cross-ref document, The "sequence" field shows if the author is the 1st, 2nd,
+     * etc. author
+     *
+     * @param sequence ordinal string e.g. "first"
+     * @return Ordinal in number format. "first" -> 1, "second" -> 2, etc.
+     */
+    private int parseSequence(String sequence, int alternativeSequence) {
+        RuleBasedNumberFormat nf = new RuleBasedNumberFormat(Locale.UK, RuleBasedNumberFormat.SPELLOUT);
+        try {
+            return nf.parse(sequence).intValue();
+        } catch (Exception e) {
+            return alternativeSequence;
+        }
+    }
+
+    private List<String> extractTags() {
```

Comment:
Set up a task to do something with the methods that return null in CrossrefConverter